cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -0,0 +1,492 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	progress_events "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"github.com/spf13/cast"
+	atlasSDK "go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+var CreateRequiredFields = []string{constants.ProjectID, constants.TenantName, constants.DataFederationRoleID, constants.DataFederationTestS3Bucket, constants.DataProcessRegion}
+var ReadRequiredFields = []string{constants.ProjectID, constants.TenantName}
+var UpdateRequiredFields = []string{constants.ProjectID, constants.TenantName, constants.DataFederationRoleID, constants.DataFederationTestS3Bucket, constants.SkipRoleValidation}
+var DeleteRequiredFields = []string{constants.ProjectID, constants.TenantName}
+var ListRequiredFields = []string{constants.ProjectID}
+
+func setup() {
+	util.SetupLogger("mongodb-atlas-data-federation")
+}
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	_, _ = logger.Debugf("Create Request: %+v\n", currentModel)
```

Comment:
i think that we should only log errors, this logs just make errors harder to find

Reply:
addressed